### DISCUSSION TITLE
gdiff output/algorithm

### DISCUSSION THREAD

**Discussion by delanym** [RsyncProject/rsync#288] [Q&A] [answered]
This paper https://www.w3.org/TR/NOTE-gdiff-19970825.html
mentions that the delta produced by rsync can be represented as a gdiff file. I don't know if this means its using the same algorithm, or its simply about capturing the difference?

I wanted to see how efficient the rsync algorithm is so I ran 
`rsync -iv --no-W a.jar b.jar`

```
sent 1,441,864 bytes  received 9,635 bytes
total size is 2,526,246  speedup is 1.74
```

Is there some way to output the gdiff for the transfer?

My end goal is to cross-check that the javaxdelta is functioning correctly and efficiently.

**Comment by WayneD** [comment, marked as answer]
One thing you can do with a default rsync is to use `--debug=deltasum2` or higher on the transfer and then sift through the info of the chunk data.  It only gives you file offsets for what matched and (for higher levels) how much literal data was received, so an additional tool would need to transform that info into some kind of diff using the original files.  You could also tweak the source in receiver.c if you wanted to make it output the file data that it is manipulating into an additional diff stream.
